src/Server.php

```diff
@@ -357,7 +356,7 @@ public function getServerUniqueId() : UuidInterface{
 	}
 
 	public function getGamemode() : GameMode{
-		return GameModeIdMap::getInstance()->fromId($this->configGroup->getConfigInt("gamemode", 0)) ?? GameMode::SURVIVAL();
+		return GameMode::fromString($this->configGroup->getConfigString("gamemode", GameMode::SURVIVAL()->name())) ?? GameMode::SURVIVAL();
```

Comment:
This will break existing configs.

Reply:
I mentioned this in my original post. Do you want to accept both ints and strings, or have some sort of migration system?

Reply:
I'm not sure. But having numeric gamemodes resolve to survival is definitely unexpected behaviour.

Reply:
I believe that something like this would work:
`[11:48:01.017] [Server thread/WARNING]: Outdated value "1" for "gamemode" property was found in server.properties. It has been migrated to "creative".`

The only thing holding that back would also be implementing it into the language files.

Reply:
Actually, strike everything I said; GameMode registers numeric aliases.